src/generators/node-config-schema/constants.mjs

```diff
@@ -0,0 +1,15 @@
+// Error Messages
+export const ERRORS = {
+  missingCCandHFiles:
+    'Both node_options.cc and node_options.h must be provided.',
+  headerTypeNotFound:
+    'A type for "{{headerKey}}" not found in the header file.',
+  missingTypeDefinition: 'No type schema found for "{{type}}".',
+};
+
+// Regex pattern to match calls to the AddOption function.
+export const ADD_OPTION_REGEX =
+  /AddOption[\s\n\r]*\([\s\n\r]*"([^"]+)"(.*?)\);/gs;
```

Comment:
Do I understand correctly that this parses the C++ source code to get the options? If so this feels like a misstep. The current C++ option parser puts an overhead at runtime to add the options. Ideally that should be refactored to add most of the options at compile time and let the parser be statically generated to eliminate the overhead and eliminate more static STL containers. Parsing the source code in an ad-hoc way in another tooling that would break the tests makes it harder to change the parser in the future. It would be better to use the binary to get the options instead of counting on the source code to be a particular shape.

Reply:
Thanks @joyeecheung for the feedback! I feel the straightforward way is [to simply use the existing code](https://github.com/nodejs/node/blob/2244a0977451c16e7cade9f1aeb46e1c136669bb/tools/doc/generate-json-schema.mjs#L4) incorporated on this repository? Or should the generation of the `node.config.json` schema simply not be handled by the `api-docs-tooling` :thinking: any thoughts or alternatives are appreciated :bow:

Reply:
Thanks Joyee, that feels like a good path forward.